src/vmware/azext_vmware/tests/latest/test_datastores_scenario.py

```diff
@@ -18,17 +18,33 @@ def setUp(self):
 
     @ResourceGroupPreparer(name_prefix='cli_test_vmware_datastores')
     def test_vmware_datastores(self):
-        # Get a pre-created iSCSI datastore
-        self.cmd('az vmware datastore show --name rasivagu-iscsi-datastore --resource-group rasivagu-cloudsan-rg --cluster Cluster-1 --private-cloud rasivagu-sddc')
-
-        # List all existing datastores
-        self.cmd('az vmware datastore list --resource-group rasivagu-cloudsan-rg --cluster Cluster-1 --private-cloud rasivagu-sddc')
+        self.kwargs.update({
+            # 'loc': 'centralus',
+            # 'privatecloud': 'cloud1',
+            # 'cluster': 'pycluster1',
+            'rg': 'rasivagu-sddc-rg',
+            'privatecloud': 'rasivagu-mock-sddc',
+            'cluster': 'Cluster-1',
+            'volume_id': '/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/ResourceGroup1/providers/Microsoft.NetApp/netAppAccounts/NetAppAccount1/capacityPools/CapacityPool1/volumes/NFSVol1',
+            'target_id': '/subscriptions/ba75e79b-dd95-4025-9dbf-3a7ae8dff2b5/resourceGroups/rasivagu-df-rg/providers/Microsoft.StoragePool/diskPools/rasivagu-df-diskpool/iscsiTargets/rasivagu-df-target'
```

Comment:
Can these resources be created dynamically instead of a fixed value？

Reply:
Not currently, no.

Reply:
May I ask why it can't be done now?

Reply:
Creating the dependent resources is not yet automated. cc @Raaghavi

Reply:
It's a little tricky to create the dependent resource dynamically. The reason for this is that `Microsoft.StoragePool` is available in PPE (Dogfood), whereas the pre-requisite resources such as disks, vnet, etc. do not have their offerings in PPE. So internally for this setup, we create the disks/vnet & underlying infrastructure for the disk pool in Public cloud, but the properties can be fetched from PPE. 
Regardless, I would like to understand the significance of creating the dependent resource dynamically. This resource is anyway not a part of the AVS resource provider. As per my understanding, the `cli` test recordings are done to check if the APIs return appropriate values in the headers/response payload. This can be achieved by testing all the AVS datastores APIs using a pre-existing StoragePool resource.
Are you thinking from a reproducibility perspective?

Reply:
Got it, thanks~
Yes, the dynamic creation of resources is to avoid the problem that the fixed resources in some environments are not authorized to access or are deleted by mistake, resulting in the test can not be repeated